pkg/runtime/kubernetes/utils.go

```diff
@@ -0,0 +1,86 @@
+// Copyright © 2022 Alibaba Group Holding Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kubernetes
+
+import (
+	"fmt"
+
+	"net"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/sealerio/sealer/common"
+	"github.com/sealerio/sealer/utils/exec"
+	osi "github.com/sealerio/sealer/utils/os"
+	"github.com/sealerio/sealer/utils/ssh"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+// VersionCompare :if v1 >= v2 return true, else return false
+func VersionCompare(v1, v2 string) bool {
+	v1 = strings.Replace(v1, "v", "", -1)
+	v2 = strings.Replace(v2, "v", "", -1)
+	v1 = strings.Split(v1, "-")[0]
+	v2 = strings.Split(v2, "-")[0]
+	v1List := strings.Split(v1, ".")
+	v2List := strings.Split(v2, ".")
+
+	if len(v1List) != 3 || len(v2List) != 3 {
+		logrus.Errorf("error version format %s %s", v1, v2)
+		return false
+	}
+	if v1List[0] > v2List[0] {
+		return true
+	} else if v1List[0] < v2List[0] {
+		return false
+	}
+	if v1List[1] > v2List[1] {
+		return true
+	} else if v1List[1] < v2List[1] {
+		return false
+	}
+	if v1List[2] > v2List[2] {
+		return true
+	}
+	return true
+}
+
+func GetKubectlAndKubeconfig(ssh ssh.Interface, host net.IP, rootfs string) error {
+	// fetch the cluster kubeconfig, and add /etc/hosts "EIP apiserver.cluster.local" so we can get the current cluster status later
+	err := ssh.Fetch(host, path.Join(common.DefaultKubeConfigDir(), "config"), common.KubeAdminConf)
+	if err != nil {
+		return errors.Wrap(err, "failed to copy kubeconfig")
+	}
+	_, err = exec.RunSimpleCmd(fmt.Sprintf("cat /etc/hosts |grep '%s %s' || echo '%s %s' >> /etc/hosts",
```

Comment:
Can we read/write file in golang instead of shell? I think it's better.

Reply:
Though I agree with you, there is many inline shell in sealer, and it is a huge project to improve it. And this func is also moved from runtime/utils.go to here.

Reply:
Got it, we can ignore this in the pr.